### PR TITLE
ENH: Support reading segmentations from NIFTI files

### PR DESCRIPTION
**Pull request by Sunderlandkyl**
Previously, segmentation files were read using vtkTeemNrrdReader.
vtkMRMLSegmentationStorage node can now read both NIFTI and NRRD file formats using vtkITKArchetypeImageSeriesVectorReaderFile.

### CODE REVIEW

File: Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx [hunk 1]
```diff
@@ -138,6 +140,7 @@ void vtkMRMLSegmentationStorageNode::InitializeSupportedReadFileTypes()
   this->SupportedReadFileTypes->InsertNextValue("Segmentation (.seg.vtm)");
   this->SupportedReadFileTypes->InsertNextValue("Segmentation (.nrrd)");
   this->SupportedReadFileTypes->InsertNextValue("Segmentation (.vtm)");
+  this->SupportedReadFileTypes->InsertNextValue("Segmentation (.nii.gz)");
```
Comment (lassoan) [inline Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx]: Add .hdr (nifti format with separate header and pixel data) and .nii (uncompressed single-file nifti).

File: Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx [hunk 2]
```diff
@@ -163,6 +166,7 @@ void vtkMRMLSegmentationStorageNode::InitializeSupportedWriteFileTypes()
     {
     this->SupportedWriteFileTypes->InsertNextValue("Segmentation (.seg.nrrd)");
     this->SupportedWriteFileTypes->InsertNextValue("Segmentation (.nrrd)");
+    this->SupportedWriteFileTypes->InsertNextValue("Segmentation (.nii.gz)");
```
Comment (lassoan) [inline Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx]: Can we write segmentation to nifti format? If we can only write labelmap without segmentation metdata then it is better not to list this format among supported write formats (it is more of a lossy export format then).

**Review by lassoan — COMMENTED** [review]
Thank you, it mostly looks good.

It would be also nice to add a test that reads data from nifti file (e.g., test import from a simple 3D labelmap stored in nifti format) and I've added a couple of comments.

**Comment by Sunderlandkyl** [comment]
Changes made.
Added vtkMRMLSegmentationStorageNodeTest1 to test loading of 3 types of segmentations.
- ITK-SNAP nii.gz
- Slicer pre shared labelmap segmentation
- Slicer post shared labelmap segmentation

**Comment by lassoan** [comment]
Thank you! Merged.

**Pull request closed without merging**